src/modules/l0-server.c

```diff
@@ -1730,6 +1730,89 @@ static bool processKeyExchange(ksnLNullClass *kl, ksnLNullData *kld, int fd,
     kld->server_crypt->state = SESCRYPT_ESTABLISHED;
     return true;
 }
+#include "jsmn.h"
+#include <openssl/md5.h>
+
+static int json_eq(const char *json, jsmntok_t *tok, const char *s) {
+	if (tok->type == JSMN_STRING && (int) strlen(s) == tok->end - tok->start &&
+			strncmp(json + tok->start, s, tok->end - tok->start) == 0) {
+		return 0;
+	}
+	return -1;
```

Comment:
Another moment that it could be something like
```
static char * json_get_string(const char *json,
                                        const jsmntok_t *tokens,
                                        int tokens_count,
                                        const char *key) {
	const int key_len = strlen(key);
	for (int i = 1; i < tokens_count; ++i) {
		const jsmntok_t *tok  = tokens + i;
  	  	const int tok_len = tok->end - tok->start;

		if (tok->type == JSMN_STRING &&
		    key_len == tok_len &&
		    strncmp(json + tok->start, s, tok_len) == 0) {
			return strndup((char*)json + tok->start,  tok_len);
		}
		return nullptr;
	}
}

```

So we could use something like this

```
    char *id = json_get_string(jsondata, tokens, r, "_id");
    char *sign = json_get_string(jsondata, tokens, r, "sign");
    char *timestamp = json_get_string(jsondata, tokens, r, "timestamp"); 
```